earthfile2llb/listener.go

```diff
@@ -434,9 +434,6 @@ func (l *listener) ExitBuildStmt(c *parser.BuildStmtContext) {
 		return
 	}
 	fullTargetName := l.expandArgs(fs.Arg(0))
-	for i, arg := range buildArgs.Args {
-		buildArgs.Args[i] = l.expandArgs(arg)
-	}
```

Comment:
This isn't right.

We definitely need to perform expansion here - not all build args are used in a shell command. So not all would be automatically expanded by the shell. For example you could have things like `FROM $SOME_BUILD_ARG` and other such commands.

Reply:
The Problem with expanding here however is, that once you do that, every ' and " will be gone. So maybe we should either use the shell expansion solely or implement the execution manually. Otherwise it might get hard to "quote" each quotes.

Reply:
I think that's intended and also, we may need to actually escape all of the `'` and `"` as you suggest. Otherwise we can't support both commands like `FROM` and also commands like `RUN` with build args.

You could use something like https://github.com/alessio/shellescape to do the heavy lifting.